gui/src/streamsession.cpp

```diff
@@ -141,66 +136,55 @@ void StreamSession::SetLoginPIN(const QString &pin)
 	chiaki_session_set_login_pin(&session, (const uint8_t *)data.constData(), data.size());
 }
 
-void StreamSession::HandleMouseEvent(QMouseEvent *event)
-{
-	if(event->type() == QEvent::MouseButtonPress)
-		keyboard_state.buttons |= CHIAKI_CONTROLLER_BUTTON_TOUCHPAD;
+void StreamSession::HandleKeyboardEvent(QKeyEvent *event) {
+	if (key_map.contains(Qt::Key(event->key())) == false)
+		return;
+
+	if (event->isAutoRepeat())
+		return;
+
+	int button = key_map[Qt::Key(event->key())];
+	bool press_event = event->type() == QEvent::Type::KeyPress;
+
+	if (press_event)
+		keyboard_state.buttons |= button;
```

Comment:
`keyboard_state.buttons` should not be touched for the extended buttons. I think this whole `if(press_event)` could be moved into the `default` case of the switch below.

Reply:
yes, you caught a bug, thanks.